tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -2024,6 +2030,46 @@ int tracepoint__sched__sched_process_fork(struct bpf_raw_tracepoint_args *ctx)
     return 0;
 }
 
+// include/trace/events/sched.h:
+//TP_PROTO(struct task_struct *p, pid_t old_pid, struct linux_binprm *bprm)
+SEC("raw_tracepoint/sched_process_exec")
+int tracepoint__sched__sched_process_exec(struct bpf_raw_tracepoint_args *ctx)
+{
+    // Note: removing container id from pid_to_cont_id_map shoule be performed before should_trace()
+    u32 pid = bpf_get_current_pid_tgid();
+    bpf_map_delete_elem(&pid_to_cont_id_map, &pid);
+
+    if (!should_trace())
+        return 0;
+
+    buf_t *submit_p = get_buf(SUBMIT_BUF_IDX);
+    if (submit_p == NULL)
+        return 0;
+    set_buf_off(SUBMIT_BUF_IDX, sizeof(context_t));
+
+    context_t context = init_and_save_context(ctx, submit_p, SCHED_PROCESS_EXEC, 2, 0);
+    struct task_struct *task = (struct task_struct *)ctx->args[0];
+    struct linux_binprm *bprm = (struct linux_binprm *)ctx->args[2];
+
+    if (bprm == NULL) {
+        return -1;
+    }
+
+    int invoked_from_kernel = has_prefix("kworker/", get_task_parent_comm(task), 9);
+    const char *filename = (const char *)READ_KERN(bprm->filename);
```

Comment:
Reading internal kernel structures may change between kernel versions.
Please add a helper function for this as well.